contracts/MechMarketplace.sol

```diff
@@ -122,44 +164,49 @@ contract MechMarketplace {
     bytes32 public immutable domainSeparator;
     // Original chain Id
     uint256 public immutable chainId;
+    // Minimum response time
+    uint256 public immutable minResponseTimeout;
+    // Maximum response time
+    uint256 public immutable maxResponseTimeout;
+    // Approved mech bytecode hash
+    bytes32 public immutable mechBytecodeHash;
     // Mech karma contract address
     address public immutable karmaProxy;
+    // Agent mech factory contract address
+    address public immutable mechStakingInstance;
 
     // Number of undelivered requests
     uint256 public numUndeliveredRequests;
     // Number of total requests
     uint256 public numTotalRequests;
-    // Minimum response time
-    uint256 public minResponseTimeout;
-    // Maximum response time
-    uint256 public maxResponseTimeout;
     // Reentrancy lock
     uint256 internal _locked = 1;
-    // Contract owner
-    address public owner;
-    // Agent mech factory contract address
-    address public factory;
 
     // Mapping of request Id => mech delivery information
     mapping(uint256 => MechDelivery) public mapRequestIdDeliveries;
     // Mapping of account nonces
     mapping(address => uint256) public mapNonces;
-    // Mapping of registered mechs
-    mapping(address => bool) public mapMechRegistrations;
 
     /// @dev MechMarketplace constructor.
-    /// @param _factory Agent mech factory address.
+    /// @param _mechStakingInstance Agent mech staking instance address.
     /// @param _karmaProxy Karma proxy contract address.
     /// @param _minResponseTimeout Min response time in sec.
     /// @param _maxResponseTimeout Max response time in sec.
-    constructor(address _factory, address _karmaProxy, uint256 _minResponseTimeout, uint256 _maxResponseTimeout) {
+    /// @param _agentMechBytecodeHash Approved agent mech bytecode hash.
+    constructor(
```

Comment:
why is factor no longer needed?

Reply:
The factory will be needed again, but not the mech factory :)